src/chains/signals.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+from urllib.parse import urljoin
+
+import requests
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+from .models import Chain
+
+logger = logging.getLogger(__name__)
+
+
+def setup_session():
+    session = requests.Session()
+    adapter = requests.adapters.HTTPAdapter(max_retries=3)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
+session = setup_session()
+
+
+@receiver(post_save, sender=Chain)
+@receiver(post_delete, sender=Chain)
+def on_chain_update(sender, **kwargs):
+    logger.info("Chain update. Triggering CGW webhook")
+    cgw_cluster_url = os.environ.get("CGW_URL")
```

Comment:
Don't use `os.environ` in a Django application. Load the environment variables on the configuration and here import `django.conf.settings` to use them. This way you prevent runtime errors